### PR TITLE
[ioctl/newcmd] modify WriteConfig to WriteAlias

### LINKED ISSUE #3144: make test error in ioctl/newcmd/account
### What version of iotex-core are you using (`ioctl version`)?
master

### What operating system and processor architecture are you using?
macOS

### What did you do? If possible, provide a recipe for reproducing the error.
`make test`

### What did you expect to see?
test pass

### What did you see instead?
```
--- FAIL: TestSign (146.53s)
    account_test.go:77: 
        	Error Trace:	account_test.go:77
        	Error:      	Received unexpected error:
        	            	open /var/folders/_y/1yg9dv650f38jfl5z12kbz_40000gn/T/ksTest/UTC--2022-03-04T01-16-27.244304000Z--371c731612f748245f24d0832d7e14dcf2b06844: no such file or directory
        	Test:       	TestSign
    panic.go:661: missing call(s) to *mock_ioctlclient.MockClient.NewKeyStore() /Users/dustin/iotexproject/iotex-core/ioctl/newcmd/account/account_test.go:63
    panic.go:661: missing call(s) to *mock_ioctlclient.MockClient.IsCryptoSm2() /Users/dustin/iotexproject/iotex-core/ioctl/newcmd/account/account_test.go:64
    panic.go:661: aborting test due to missing call(s)
FAIL
FAIL	github.com/iotexproject/iotex-core/ioctl/newcmd/account	528.418s
```
however, if I go into the folder to run tests it can pass, like below
1. `cd ioctl/newcmd/account`
2. `go test ./...`
i got test pass

### PR DESCRIPTION
**Pull request by huof6829**
fix #3144

### CODE REVIEW

**Review by CoderZhi — APPROVED** [review]

File: ioctl/client.go [hunk 1]
```diff
@@ -71,9 +73,10 @@ type (
 	}
 
 	client struct {
-		cfg       config.Config
-		conn      *grpc.ClientConn
-		cryptoSm2 bool
+		cfg               config.Config
+		conn              *grpc.ClientConn
+		cryptoSm2         bool
+		defaultConfigFile string
```
[the remark this answers is not in the document]
Comment (huof6829) [reply ioctl/client.go]: ok

File: ioctl/client.go [hunk 2]
```diff
@@ -235,13 +239,28 @@ func (c *client) AliasMap() map[string]string {
 	return aliases
 }
 
-func (c *client) WriteConfig(cfg config.Config) error {
-	out, err := yaml.Marshal(&cfg)
+func (c *client) SetAlias(alias string, addr string) error {
+	aliases := c.AliasMap()
+	for aliases[addr] != "" {
```
Comment (Liuhaai) [inline ioctl/client.go:244]: Don't use `for {}` here

Comment (huof6829) [reply ioctl/client.go:244]: see  https://github.com/iotexproject/iotex-core/blob/182c040a6c7a7922965ab4679b072f29ccd941b1/ioctl/cmd/alias/aliasset.go#L56

Comment (Liuhaai) [reply ioctl/client.go:244]: > see
> 
> https://github.com/iotexproject/iotex-core/blob/182c040a6c7a7922965ab4679b072f29ccd941b1/ioctl/cmd/alias/aliasset.go#L56

You could improve it in this pr

**Review by Liuhaai — APPROVED** [review]

File: ioctl/client.go (hunk 2 above)
Comment (Liuhaai) [reply ioctl/client.go:244]: Could you help fix https://github.com/iotexproject/iotex-core/issues/3171 when free?

Comment (huof6829) [reply ioctl/client.go:244]: ok

**Pull request merged**